scripts/helper_scripts/new-parsers/src/taxons_uniprots_tables/models.rs

```diff
@@ -0,0 +1,183 @@
+use std::fmt;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+#[derive(Debug)]
+pub struct Entry {
+    pub min_length: u32,
+    pub max_length: u32,
+
+    // These three are actually ints, but sthey are never used as ints,
+    // so there is no use converting/parsing them as such
+    pub accession_number: String,
+    pub version: String,
+    pub taxon_id: i32,
+
+    pub type_: String,
+    pub name: String,
+    pub sequence: String,
+    pub ec_references: Vec<String>,
+    pub go_references: Vec<String>,
+    pub ip_references: Vec<String>,
+}
+
+impl Entry {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        min_length: u32,
+        max_length: u32,
+        type_: String,
+        accession_number: String,
+        sequence: String,
+        name: String,
+        version: String,
+        taxon_id: String,
+    ) -> Self {
+        Entry {
+            min_length,
+            max_length,
+
+            accession_number,
+            version,
+            taxon_id: taxon_id.parse().unwrap(),
+            type_,
+            name,
+            sequence,
+
+            ec_references: Vec::new(),
+            go_references: Vec::new(),
+            ip_references: Vec::new(),
+        }
+    }
+
+    pub fn digest(&mut self) -> Vec<String> {
+        let mut result = Vec::new();
+
+        let mut start: usize = 0;
+        let length = self.sequence.len();
+        let content = self.sequence.as_bytes();
+
+        for (i, c) in content.iter().enumerate() {
+            if (*c == b'K' || *c == b'R') && (i + 1 < length && content[i + 1] != b'P') {
+                if i + 1 - start >= self.min_length as usize
+                    && i + 1 - start <= self.max_length as usize
+                {
+                    result.push(String::from_utf8_lossy(&content[start..i + 1]).to_string());
+                }
+
+                start = i + 1;
+            }
+        }
+
+        // Add last one
+        if length - start >= self.min_length as usize && length - start <= self.max_length as usize
+        {
+            result.push(String::from_utf8_lossy(&content[start..length]).to_string())
+        }
+
+        result
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct RankParseError {
+    input: String,
+}
+
+impl fmt::Display for RankParseError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "unable to parse {} as Rank", self.input)
+    }
+}
+
+#[derive(Debug)]
+pub enum Rank {
```

Comment:
Umgap's rank contains 2 entries that the current Java tool doesn't (`Infraclass` and `Parvorder`). I don't know if this could be breaking or not, so I can't make the decision to replace it.

https://github.com/unipept/unipept-database/blob/master/scripts/helper_scripts/parser/src/taxons/Taxon.java#L31

Reply:
Right; I guess a copy is better for now, then. It could be unified if I ever get to update the ranks in UMGAP.

Reply:
@stijndcl The Unipept taxonomy was updated two years ago since NCBI also updated theirs. We decided to get rid of Infraclass and Parvorder (since almost no peptides where annotated with an LCA at these ranks anyways) and we've added the new `strain` rank. If you want more information about, you can check the discussion in this PR: https://github.com/unipept/unipept-database/pull/17